components/admin-panel/sections/ActivityLog/ActivityAccountFilter.js

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useRouter } from 'next/router';
+
+import { CollectiveType } from '../../../../lib/constants/collectives';
+
+import { CUSTOM_OPTIONS_POSITION, FLAG_COLLECTIVE_PICKER_COLLECTIVE } from '../../../CollectivePicker';
+import CollectivePickerAsync from '../../../CollectivePickerAsync';
+
+const ActivityAccountFilter = ({ account, onChange }) => {
+  const router = useRouter();
+  const defaultValue = {
+    value: { id: account?.legacyId, name: account?.name, slug: account?.slug, imageUrl: account?.imageUrl },
+    label: account?.name,
+    [FLAG_COLLECTIVE_PICKER_COLLECTIVE]: true,
+  };
+  const [selectedValue, setSelectedValue] = React.useState(defaultValue);
+
+  useEffect(() => {
+    if (router?.query?.account) {
+      router.push(`${account.slug}/admin/activity-log`);
+    }
+  }, []);
```

Comment:
For now I've made sure that if the user directly comes to a url with a collective filter we just redirect to the default filter. Do you think we should support this at this point? Didn't want to make this too complicated by running an additional query to obtain the collective etc. 🤔

Reply:
This won't work with the root dashboard (where the activity log is also used) and overall I don't think we need to redirect, just show the proper option(s) in the select